testrace: sql: error in TestLeaseManager

Found in https://teamcity.cockroachdb.com/viewLog.html?buildId=313133&

```
[TestLeaseManager] [Test Output]
W170802 17:39:49.913435 76476 server/server.go:298  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170802 17:39:49.915216 76476 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170802 17:39:49.928301 76476 server/config.go:521  [n?] 1 storage engine initialized
I170802 17:39:49.928359 76476 server/config.go:523  [n?] RocksDB cache size: 512 MiB
I170802 17:39:49.928398 76476 server/config.go:523  [n?] store 0: in-memory, size 100 MiB
I170802 17:39:49.952548 76476 server/node.go:335  [n?] **** cluster 3c298d78-fec4-44d4-8648-5d71a82a87ba has been created
I170802 17:39:49.952645 76476 server/server.go:780  [n?] **** add additional nodes by specifying --join=127.0.0.1:43477
I170802 17:39:49.972954 76476 storage/store.go:1271  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170802 17:39:49.984663 76476 server/node.go:421  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:1 WritesPerSecond:65.79958387685161}
I170802 17:39:49.984924 76476 server/node.go:319  [n1] node ID 1 initialized
I170802 17:39:49.985886 76476 gossip/gossip.go:295  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:43477" > attrs:<> locality:<> 
I170802 17:39:49.986452 76476 storage/stores.go:293  [n1] read 0 node addresses from persistent storage
I170802 17:39:49.986653 76476 server/node.go:556  [n1] connecting to gossip network to verify cluster ID...
I170802 17:39:49.989110 76476 server/node.go:581  [n1] node connected via gossip and verified as part of cluster "3c298d78-fec4-44d4-8648-5d71a82a87ba"
I170802 17:39:49.989433 76476 server/node.go:362  [n1] node=1: started with [=] engine(s) and attributes []
I170802 17:39:49.991697 76476 sql/executor.go:423  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:43477}
I170802 17:39:50.045241 76476 server/server.go:859  [n1] starting https server at 127.0.0.1:34442
I170802 17:39:50.049343 76476 server/server.go:860  [n1] starting grpc/postgres server at 127.0.0.1:43477
I170802 17:39:50.049444 76476 server/server.go:861  [n1] advertising CockroachDB node at 127.0.0.1:43477
W170802 17:39:50.049821 76476 sql/jobs/registry.go:148  [n1] unable to get node liveness: node not in the liveness table
W170802 17:39:50.049895 76476 sql/jobs/registry.go:273  [n1] canceling all jobs due to liveness failure
I170802 17:39:50.051282 76616 storage/replica_command.go:2667  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170802 17:39:50.253278 76616 storage/replica_command.go:2667  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/tsd [r3]
E170802 17:39:50.254965 76621 storage/consistency_queue.go:90  [replica consistency checker,n1,s1,r1/1:/{Min-System/}] key range /Min-/Max outside of bounds of range /Min-/System/""
E170802 17:39:50.309120 76412 storage/replica_proposal.go:538  [n1,s1,r2/1:/{System/-Max}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
I170802 17:39:50.314185 76616 storage/replica_command.go:2667  [split,n1,s1,r3/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r4]
E170802 17:39:50.397946 76432 storage/replica_proposal.go:538  [n1,s1,r3/1:/{System/tsd-Max}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
I170802 17:39:50.404536 76616 storage/replica_command.go:2667  [split,n1,s1,r4/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r5]
I170802 17:39:50.410022 76476 sql/event_log.go:99  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170802 17:39:50.485134 76616 storage/replica_command.go:2667  [split,n1,s1,r5/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r6]
I170802 17:39:50.496782 76476 sql/lease.go:341  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-08-02 17:39:50.49303006 +0000 UTC
I170802 17:39:50.560242 76616 storage/replica_command.go:2667  [split,n1,s1,r6/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r7]
I170802 17:39:50.673161 76616 storage/replica_command.go:2667  [split,n1,s1,r7/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r8]
I170802 17:39:50.741034 76616 storage/replica_command.go:2667  [split,n1,s1,r8/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r9]
I170802 17:39:50.817570 76476 server/server.go:999  [n1] done ensuring all necessary migrations have run
I170802 17:39:50.817689 76476 server/server.go:1001  [n1] serving sql connections
I170802 17:39:50.857102 76940 sql/event_log.go:99  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:43477} Attrs: Locality:} ClusterID:3c298d78-fec4-44d4-8648-5d71a82a87ba StartedAt:1501695589989159047 LastUp:1501695589989159047}
I170802 17:39:50.884875 76616 storage/replica_command.go:2667  [split,n1,s1,r9/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r10]
I170802 17:39:50.961251 76616 storage/replica_command.go:2667  [split,n1,s1,r10/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r11]
I170802 17:39:51.079758 76616 storage/replica_command.go:2667  [split,n1,s1,r11/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r12]
I170802 17:39:51.174169 76616 storage/replica_command.go:2667  [split,n1,s1,r12/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r13]
I170802 17:39:51.229277 76616 storage/replica_command.go:2667  [split,n1,s1,r13/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r14]
I170802 17:39:51.480094 76476 sql/lease.go:341  publish: descID=11 (lease) version=2 mtime=2017-08-02 17:39:51.475356449 +0000 UTC
I170802 17:39:51.530112 76476 util/stop/stopper.go:471  quiescing; tasks left:
1      node.Node: batch
1      [async] sql.tableState: releasing descriptor lease
I170802 17:39:51.535325 76476 util/stop/stopper.go:471  quiescing; tasks left:
1      [async] sql.tableState: releasing descriptor lease
W170802 17:39:51.535991 77289 sql/lease.go:228  error releasing lease "11(\"lease\") ver=1:1501695853.949064358,0, refcount=0": node unavailable; try another peer
I170802 17:39:51.536867 76738 kv/transport_race.go:69  transport race promotion: ran 19 iterations on up to 431 requests
[17:44:10][TestLeaseManager] lease_test.go:105: expected /1/1 /2/1, but found /1/1 /1/1 /2/1
```

cc @cockroachdb/sql-async

--- a/pkg/sql/lease_test.go
+++ b/pkg/sql/lease_test.go
@@ -100,10 +100,13 @@ SELECT version, "nodeID" FROM system.lease WHERE "descID" = $1 ORDER BY version,
 }
 
 func (t *leaseTest) expectLeases(descID sqlbase.ID, expected string) {
-	leases := t.getLeases(descID)
-	if expected != leases {
-		t.Fatalf("expected %s, but found %s", expected, leases)
-	}
+	testutils.SucceedsSoon(t, func() error {
+		leases := t.getLeases(descID)
+		if expected != leases {
+			return errors.Errorf("expected %s, but found %s", expected, leases)
+		}
+		return nil
+	})
 }
 
 func (t *leaseTest) acquire(
@@ -238,6 +241,7 @@ func TestLeaseManager(testingT *testing.T) {
 	if _, _, err := t.acquireMinVersion(1, descID, 2); !testutils.IsError(err, expected) {
 		t.Fatalf("expected %s, but found %v", expected, err)
 	}
+	t.expectLeases(descID, "/1/1")
 
 	// Publish a new version and explicitly acquire it.
 	l2, _ = t.mustAcquire(1, descID)

